File: internal/content/testdata/bench/javascript/nc-valid-anagram.js
function is_anagram(s, t) {
  if (s.length !== t.length) return false;
  const counts = {};
  for (const c of s) {
    counts[c] = (counts[c] || 0) + 1;
  }
  for (const c of t) {
    if (!counts[c]) return false;
    counts[c]--;
  }
  return true;
}

File: internal/content/testdata/bench/csharp/nc-valid-anagram.cs
public class Solution {
    public bool is_anagram(string s, string t) {
        if (s.Length != t.Length) return false;
        var counts = new Dictionary<char, int>();
        foreach (char c in s) {
            if (!counts.ContainsKey(c)) counts[c] = 0;
            counts[c]++;
        }
        foreach (char c in t) {
            if (!counts.ContainsKey(c) || counts[c] == 0) return false;
            counts[c]--;
        }
        return true;
    }
}
